test/widgets/compose_box_test.dart

```diff
@@ -405,97 +407,249 @@ void main() {
+            streams: [eg.stream(streamId: 1, channelPostPolicy: policy)]);
+          checkComposeBox(isShown: canPost);
+        });
+      }
+
+      group('only "full member" user can post in channel with "fullMembers" policy', () {
+        testWidgets('"full member" -> can post in channel', (tester) async {
+          final selfUser = eg.user(role: UserRole.member,
+            dateJoined: DateTime.now().subtract(const Duration(days: 30)).toIso8601String());
+          await prepareComposeBox(tester,
+            narrow: const ChannelNarrow(1),
+            selfUser: selfUser,
+            realmWaitingPeriodThreshold: 30,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: ChannelPostPolicy.fullMembers)]);
+          checkComposeBox(isShown: true);
+        });
+
+        testWidgets('not a "full member" -> cannot post in channel', (tester) async {
+          final selfUser = eg.user(role: UserRole.member,
+            dateJoined: DateTime.now().subtract(const Duration(days: 29)).toIso8601String());
+          await prepareComposeBox(tester,
+            narrow: const ChannelNarrow(1),
+            selfUser: selfUser,
+            realmWaitingPeriodThreshold: 30,
+            streams: [eg.stream(streamId: 1, channelPostPolicy: ChannelPostPolicy.fullMembers)]);
+          checkComposeBox(isShown: false);
+        });
       });
 
-      testWidgets('at least one user becomes deactivated -> '
-          'compose box is replaced with a banner', (tester) async {
-        final activeUsers = [eg.user(isActive: true), eg.user(isActive: true)];
-        await prepareComposeBox(tester, narrow: groupDmNarrowWith(activeUsers),
-          users: activeUsers);
+      Future<void> changeUserRole(WidgetTester tester,
+          {required User user, required UserRole role}) async {
+        await store.handleEvent(RealmUserUpdateEvent(id: 1,
+          userId: user.userId, role: role));
+        await tester.pump();
+      }
+
+      Future<void> changeChannelPolicy(WidgetTester tester,
+          {required ZulipStream channel, required ChannelPostPolicy policy}) async {
+        await store.handleEvent(eg.channelUpdateEvent(channel,
+          property: ChannelPropertyName.channelPostPolicy, value: policy));
+        await tester.pump();
+      }
+
+      testWidgets('user role decreases -> compose box is replaced with the banner', (tester) async {
+        final selfUser = eg.user(role: UserRole.administrator);
+        await prepareComposeBox(tester,
+          narrow: const ChannelNarrow(1),
+          selfUser: selfUser,
+          streams: [eg.stream(streamId: 1, channelPostPolicy: ChannelPostPolicy.administrators)]);
         checkComposeBox(isShown: true);
 
-        await changeUserStatus(tester, user: activeUsers[0], isActive: false);
+        await changeUserRole(tester, user: selfUser, role: UserRole.moderator);
         checkComposeBox(isShown: false);
       });
 
-      testWidgets('all deactivated users become active -> '
```

Comment:
Hmm, the deletion is a bit far apart from where it was re-added.

A goal of restructuring the change would be that we have one commit that moves the old tests, and one that only adds the new ones.